From dbc1b5c42e563f5da1307d83a7aeb21b26594eff Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 27 Oct 2021 14:56:51 +0200
Subject: [PATCH] binding: use -std=c++14

:Release Notes:

:Detailed Notes:
DEBUG: Executing shell function do_compile
gyp info it worked if it ends with ok
gyp info using node-gyp@6.1.0
gyp info using node@16.11.1 | linux | x64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/git/build'
  ACTION Generating trusted scripts list src/trusted_scripts.inc
  CXX(target) Release/obj.target/webos-sysbus/src/node_ls2.o
In file included from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/deps/v8/include/v8.h:30,
                 from /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/src/node.h:63,
                 from ../src/node_ls2.cpp:19:
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/deps/v8/include/v8-internal.h: In function 'void v8::internal::PerformCastCheck(T*)':
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/deps/v8/include/v8-internal.h:492:38: error: 'remove_cv_t' is not a member of 'std'; did you mean 'remove_cv'?
  492 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                      ^~~~~~~~~~~
      |                                      remove_cv
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/deps/v8/include/v8-internal.h:492:38: error: 'remove_cv_t' is not a member of 'std'; did you mean 'remove_cv'?
  492 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                      ^~~~~~~~~~~
      |                                      remove_cv
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/deps/v8/include/v8-internal.h:492:50: error: template argument 2 is invalid
  492 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                                  ^
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/deps/v8/include/v8-internal.h:492:63: error: '::Perform' has not been declared
  492 |             !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
      |                                                               ^~~~~~~
In file included from ../src/node_ls2.cpp:19:
../src/node_ls2.cpp: At global scope:
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/src/node.h:821:7: warning: cast between incompatible function types from 'void (*)(v8::Local<v8::Object>)' to 'node::addon_register_func' {aka 'void (*)(v8::Local<v8::Object>, v8::Local<v8::Value>, void*)'} [-Wcast-function-type]
  821 |       (node::addon_register_func) (regfunc),                          \
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/node-v16.11.1/src/node.h:855:3: note: in expansion of macro 'NODE_MODULE_X'
  855 |   NODE_MODULE_X(modname, regfunc, NULL, 0)  // NOLINT (readability/null_usage)
      |   ^~~~~~~~~~~~~
../src/node_ls2.cpp:282:1: note: in expansion of macro 'NODE_MODULE'
  282 | NODE_MODULE(webos_sysbus, init)
      | ^~~~~~~~~~~
make: *** [webos-sysbus.target.mk:136: Release/obj.target/webos-sysbus/src/node_ls2.o] Error 1
make: Leaving directory '/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/git/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/recipe-sysroot-native/usr/lib/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:390:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Linux 5.13.0-20-generic
gyp ERR! command "/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/recipe-sysroot-native/usr/bin/node" "/OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/recipe-sysroot-native/usr/bin/node-gyp" "--arch" "x86_64" "build"
gyp ERR! cwd /OE/build/luneos-kirkstone/webos-ports/tmp-glibc/work/core2-64-webos-linux/nodejs-module-webos-sysbus/3.0.1-1+gitAUTOINC+db4c888e62-r0/git
gyp ERR! node -v v16.11.1
gyp ERR! node-gyp -v v6.1.0
gyp ERR! not ok
WARNING: exit code 1 from a shell command.

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-310] Apply .patch files for hot-fix to corresponding repositories
[WRP-10847] Fix LIC_FILES_CHKSUM usage in open-source recipes
[WRP-10791] CCC: Various build fixes

Change-Id: I881f9092e693a2b45fae4a2dd9e087701d995457
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/nodejs-module-webos-sysbus/+/348069 binding: use -std=c++14]

 CMakeLists.txt | 2 +-
 binding.gyp    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ebc0de..b09f865 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,7 +36,7 @@ pkg_check_modules(GLIB2 REQUIRED glib-2.0)
 include_directories(${GLIB2_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${LSGLIB2_CFLAGS_OTHER})
 
-webos_add_compiler_flags(ALL -g -Wall -Wno-error=strict-aliasing -DEV_MULTIPLICITY=0 CXX -std=c++11)
+webos_add_compiler_flags(ALL -g -Wall -Wno-error=strict-aliasing -DEV_MULTIPLICITY=0 CXX -std=c++14)
 
 # Can't specify --no-undefined because the plugin is allowed to link with all of
 # the routines from the embedded v8 when it is loaded at runtime, but there's no
diff --git a/binding.gyp b/binding.gyp
index 9ff3d7a..2d31089 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -40,7 +40,7 @@
       },
       'cflags!': [ '-fno-exceptions' ],
       'cflags': [ '-g' ],
-      'cflags_cc': [ '-g', '--std=c++11' ],
+      'cflags_cc': [ '-g', '--std=c++14' ],
       'cflags_cc!': [ '-fno-exceptions' ],
       'ldflags': [ '-pthread' ],
       'actions': [
